pkg/handlers/v1/cloud_fetch.go

```diff
@@ -117,44 +164,113 @@ func (h *CloudFetchByHostnameHandler) Handle(ctx context.Context, input CloudAss
 	return extractOutput(assets), nil
 }
 
+func validateAssetType(input string) (string, error) {
+	switch input {
+	case awsEC2, awsELB, awsALB:
+		return input, nil
+	default:
+		return "", fmt.Errorf("unknown asset type %s", input)
+	}
+}
+
 // CloudFetchAllAssetsByTimeHandler defines a lambda handler for bulk fetching cloud assets known at specific point in time
 type CloudFetchAllAssetsByTimeHandler struct {
 	LogFn   domain.LogFn
 	StatFn  domain.StatFn
 	Fetcher domain.CloudAllAssetsByTimeFetcher
 }
 
-// Handle handles fetching cloud assets by hostname
-func (h *CloudFetchAllAssetsByTimeHandler) Handle(ctx context.Context, input CloudAssetFetchAllByTimestampParameters) (CloudAssets, error) {
+// Handle handles fetching cloud assets with pagination
+func (h *CloudFetchAllAssetsByTimeHandler) Handle(ctx context.Context, input CloudAssetFetchAllByTimestampParameters) (PagedCloudAssets, error) {
 	logger := h.LogFn(ctx)
 
 	ts, e := time.Parse(time.RFC3339Nano, input.Timestamp)
 	if e != nil {
 		logger.Info(logs.InvalidInput{Reason: e.Error()})
-		return CloudAssets{}, InvalidInput{Field: "time", Cause: e}
+		return PagedCloudAssets{}, InvalidInput{Field: "time", Cause: e}
 	}
 
-	if input.Count == nil {
+	if input.Count == 0 {
 		e = errors.New("missing or malformed required parameter count")
 		logger.Info(logs.InvalidInput{Reason: e.Error()})
-		return CloudAssets{}, InvalidInput{Field: "count", Cause: e}
+		return PagedCloudAssets{}, InvalidInput{Field: "count", Cause: e}
 	}
 
-	var offset uint
-	if input.Offset != nil {
-		offset = *input.Offset
+	assetType, e := validateAssetType(input.Type)
+	if e != nil {
+		logger.Info(logs.InvalidInput{Reason: e.Error()})
+		return PagedCloudAssets{}, InvalidInput{Field: "type", Cause: e}
 	}
 
-	assets, e := h.Fetcher.FetchAll(ctx, ts, *input.Count, offset)
+	var offset uint // do not offset as this is the first page
+	assets, e := h.Fetcher.FetchAll(ctx, ts, input.Count, offset, assetType)
 	if e != nil {
 		logger.Error(logs.StorageError{Reason: e.Error()})
-		return CloudAssets{}, e
+		return PagedCloudAssets{}, e
 	}
 	if len(assets) == 0 {
-		return CloudAssets{}, NotFound{ID: "any"}
+		return PagedCloudAssets{}, NotFound{ID: "any"}
```

Comment:
suggestion: This was not introduced in this PR, but we would usually keep custom error types in the `domain` package since they are used to communicate across interface boundaries.